code/modules/mob/living/silicon/login.dm

```diff
@@ -10,6 +10,8 @@
 /mob/living/silicon/auto_deadmin_on_login()
 	if(!client?.holder)
 		return TRUE
-	if(CONFIG_GET(flag/auto_deadmin_silicons) || (client.prefs?.toggles & DEADMIN_POSITION_SILICON))
+	if(!istype(src, /mob/living/silicon/pai) && (CONFIG_GET(flag/auto_deadmin_silicons) || (client.prefs?.toggles & DEADMIN_POSITION_SILICON)))
+		return client.holder.auto_deadmin()
+	if(istype(src, /mob/living/silicon/pai) && (CONFIG_GET(flag/auto_deadmin_pai) || (client.prefs?.toggles & DEADMIN_POSITION_PAI)))
```

Comment:
```suggestion
	if(istype(src, /mob/living/silicon/pai) && (CONFIG_GET(flag/auto_deadmin_pai) || (client.prefs?.toggles & DEADMIN_POSITION_PAI)))
		return client.holder.auto_deadmin()
	if(CONFIG_GET(flag/auto_deadmin_silicons) || (client.prefs?.toggles & DEADMIN_POSITION_PAI)))
```

Reply:
tbh, this logic is still pretty yucky.  Two checks for the same return is kinda nasty.